src/util/orm.py

```diff
@@ -719,7 +719,10 @@ def _add_magic_fields(self):
 
     with all_patches():
         # 4. Reload the registry with the models and fields converted to manual.
-        env.registry.setup_models(env.cr)
+        if version_gte("saas~18.2"):
+            env.registry._setup_models__(env.cr)
+        else:
+            env.registry.setup_models(env.cr)
```

Comment:
What about using `hasattr` instead of comparing version?
```suggestion
        setup_models = env.registry._setup_models__ if hasattr(env.registry, "_setup_models__") else env.registry.setup_models
        setup_models(env.cr)
```